pkg/formatted/color.go

```diff
@@ -16,19 +16,64 @@ package formatted
 
 import (
 	"io"
+	"sync"
 
 	"github.com/fatih/color"
 )
 
+var (
+	mux       sync.Mutex
+	allColors = []color.Attribute{
+		color.FgHiGreen,
+		color.FgHiYellow,
+		color.FgHiBlue,
+		color.FgHiMagenta,
+		color.FgHiCyan,
+		color.FgHiWhite,
+	}
+)
+
+type RainbowColors struct {
+	current int
+	cached  map[string]color.Attribute
+	all     []color.Attribute
+}
+
+func NewRainbowColors() *RainbowColors {
+	return &RainbowColors{
+		current: 0,
+		cached:  make(map[string]color.Attribute),
+		all:     allColors,
+	}
+}
+
+func (rc *RainbowColors) Get(label string) *color.Color {
+	mux.Lock()
```

Comment:
mux must be part of the `rc` and not a package level variable; this allows for mulitple RainbowColors to be used without having one instance to wait for another instance.